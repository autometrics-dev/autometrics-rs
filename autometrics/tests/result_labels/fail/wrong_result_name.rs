// This test ensures that the macro fails with a readable
// error when the attribute given to one variant inside the
// enumeration does not use the correct key for the label.
use autometrics_macros::ResultLabels;

struct Inner {}

#[derive(ResultLabels)]
enum MyError {
    Empty,
    #[label(unknown = "ok")]
    ClientError {
        inner: Inner,
    },
    ServerError(u64),
}

fn main() {}
