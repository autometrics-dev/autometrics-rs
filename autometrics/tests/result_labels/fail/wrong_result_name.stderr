error: Only `result = "RES"` (RES can be "ok" or "error") is supported
  --> tests/result_labels/fail/wrong_result_name.rs:11:13
   |
11 |     #[label(unknown = "ok")]
   |             ^^^^^^^
